x/rewards/keeper/iprpc.go

```diff
@@ -17,49 +17,56 @@ func (k Keeper) FundIprpc(ctx sdk.Context, creator string, duration uint64, fund
 		return utils.LavaFormatWarning("spec not found or disabled", types.ErrFundIprpc)
 	}
 
-	// check fund consists of minimum amount of ulava (min_iprpc_cost)
-	minIprpcFundCost := k.GetMinIprpcCost(ctx)
-	if fund.AmountOf(k.stakingKeeper.BondDenom(ctx)).LT(minIprpcFundCost.Amount) {
-		return utils.LavaFormatWarning("insufficient ulava tokens in fund. should be at least min iprpc cost * duration", types.ErrFundIprpc,
-			utils.LogAttr("min_iprpc_cost", k.GetMinIprpcCost(ctx).String()),
-			utils.LogAttr("duration", strconv.FormatUint(duration, 10)),
-			utils.LogAttr("fund_ulava_amount", fund.AmountOf(k.stakingKeeper.BondDenom(ctx))),
-		)
-	} else if fund.IsEqual(sdk.NewCoins(minIprpcFundCost)) {
-		return utils.LavaFormatWarning("funds are equal to min iprpc cost, no funds left to send to iprpc pool", types.ErrFundIprpc,
-			utils.LogAttr("creator", creator),
-			utils.LogAttr("spec", spec),
-			utils.LogAttr("funds", fund.String()),
-			utils.LogAttr("min_iprpc_cost", minIprpcFundCost.String()),
-		)
-	}
-
-	// check creator has enough balance
-	addr, err := sdk.AccAddressFromBech32(creator)
-	if err != nil {
-		return utils.LavaFormatWarning("invalid creator address", types.ErrFundIprpc)
-	}
+	// calculate total funds to transfer to the IPRPC pool (input fund is monthly fund)
+	totalFunds := fund.MulInt(math.NewIntFromUint64(duration))
+
+	// if the fund TX originates from the gov module (keeper's authority field) it's not paying the minimum IPRPC cost
```

Comment:
update the comment also for types.IbcIprpcReceiver

Reply:
outdated. Resolving.